### PR TITLE
Adds protector holoparasites to traitor

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Adds protector holoparasites to traitor holoparasite injectors.
rscadd: Protector holoparasites cause the summoner to teleport to them when out of range, instead of the other way around.
rscadd: Protector holoparasites have two modes; Combat, where they do and take medium damage, and Protection, where they do and take almost no damage, but move slightly slower.
/:cl:

alwaysuseprotection

### CODE REVIEW

**Comment by Bawhoppen** [comment]
lotsa new holoparasites, don't drive this feature to death pls

**Comment by Lumipharon** [comment]
This makes the owner very literally unkillable without ranged lethal weapons.

File: code/modules/mob/living/simple_animal/guardian/types/protector.dm [hunk 1]
```diff
@@ -2,10 +2,57 @@
 /mob/living/simple_animal/hostile/guardian/protector
 	melee_damage_lower = 15
 	melee_damage_upper = 15
-	damage_coeff = list(BRUTE = 0.3, BURN = 0.3, TOX = 0.3, CLONE = 0.3, STAMINA = 0, OXY = 0.3)
-	playstyle_string = "<span class='holoparasite'>As a <b>protector</b> type you have very high damage resistance, a decent attack, and cause your summoner to leash to you instead of you leashing to them.</span>"
+	range = 15 //worse for it due to how it leashes
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/guardian/types/protector.dm]: This is high enough that it will not trigger in the thunderdome, for reference.

**Comment by Jordie0608** [comment]
> where they do and take medium damage
where they do and take almost no damage

Do what?

**Comment by Bawhoppen** [comment]
I think he mean damage, they take medium damage and deal medium damage in one mode, and take and deal little in another mode

**Comment by Armhulen** [comment]
I honestly think this should have a nerf, and really, should balance out the other ones before doing it. I love more holos, but I don't want a time where one holo is always better than the rest

**Comment by ChangelingRain** [comment]
What sort of nerf? And are you sure it's going to be absolutely 100% the best choice every time?
Please be specific when complaining about possible issues

**Pull request merged**